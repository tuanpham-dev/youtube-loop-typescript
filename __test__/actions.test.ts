import configureMockStore from 'redux-mock-store'
import {
  addVideoByYouTubeId, addVideo, removeVideo,
  editVideo, updateVideos, playVideo,
  playNextVideo, playPreviousVideo, stopVideo
} from '../src/store/actions'
import {
  ADD_VIDEO_BY_YOUTUBE_ID, ADD_VIDEO, REMOVE_VIDEO,
  EDIT_VIDEO, UPDATE_VIDEOS, PLAY_VIDEO,
  PLAY_PREVIOUS_VIDEO, PLAY_NEXT_VIDEO
} from '../src/store/types'
import mockData from './mock-data/video'

const mockStore = configureMockStore()
let store = mockStore({})

describe('actions', () => {
  it('should create ADD_VIDEO_BY_YOUTUBE_ID when adding video by youtube id', () => {
    store = mockStore({})
    store.dispatch(addVideoByYouTubeId(mockData.youtubeId))

    const actions = store.getActions()
    const expectedPayload = {
      type: ADD_VIDEO_BY_YOUTUBE_ID,
      payload: mockData.youtubeId
    }
    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create ADD_VIDEO when adding a video', () => {
    store = mockStore({})
    store.dispatch(addVideo(mockData.video))

    const actions = store.getActions()
    const expectedPayload = {
      type: ADD_VIDEO,
      payload: mockData.video
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create REMOVE_VIDEO when removing a video', () => {
    store = mockStore({})
    store.dispatch(removeVideo(mockData.video.id))

    const actions = store.getActions()
    const expectedPayload = {
      type: REMOVE_VIDEO,
      payload: mockData.video.id
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create EDIT_VIDEO when editing a video', () => {
    store = mockStore({})
    store.dispatch(editVideo(mockData.video))

    const actions = store.getActions()
    const expectedPayload = {
      type: EDIT_VIDEO,
      payload: mockData.video
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create UPDATE_VIDEOS when updating videos', () => {
    store = mockStore({})
    store.dispatch(updateVideos(mockData.videos))

    const actions = store.getActions()
    const expectedPayload = {
      type: UPDATE_VIDEOS,
      payload: mockData.videos
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create PLAY_VIDEO when playing a video', () => {
    store = mockStore({})
    store.dispatch(playVideo(mockData.playingVideo))

    const actions = store.getActions()
    const expectedPayload = {
      type: PLAY_VIDEO,
      payload: mockData.playingVideo
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create PLAY_VIDEO when stopping a video', () => {
    store = mockStore({})
    store.dispatch(stopVideo())

    const actions = store.getActions()
    const expectedPayload = {
      type: PLAY_VIDEO,
      payload: null
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create PLAY_NEXT_VIDEO when playing previous video', () => {
    store = mockStore({})
    store.dispatch(playPreviousVideo())

    const actions = store.getActions()
    const expectedPayload = {
      type: PLAY_PREVIOUS_VIDEO
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })

  it('should create PLAY_NEXT_VIDEO when playing next video', () => {
    store = mockStore({})
    store.dispatch(playNextVideo())

    const actions = store.getActions()
    const expectedPayload = {
      type: PLAY_NEXT_VIDEO
    }

    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })
})
