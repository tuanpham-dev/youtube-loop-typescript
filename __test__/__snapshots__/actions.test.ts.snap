// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create ADD_VIDEO when adding a video 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 1,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
    "type": "ADD_VIDEO",
  },
]
`;

exports[`actions should create ADD_VIDEO_BY_YOUTUBE_ID when adding video by youtube id 1`] = `
Array [
  Object {
    "payload": "nfWlot6h_JM",
    "type": "ADD_VIDEO_BY_YOUTUBE_ID",
  },
]
`;

exports[`actions should create EDIT_VIDEO when editing a video 1`] = `
Array [
  Object {
    "payload": Object {
      "id": 1,
      "range": Array [
        0,
        230,
      ],
      "volume": 50,
      "youtubeId": "nfWlot6h_JM",
    },
    "type": "EDIT_VIDEO",
  },
]
`;

exports[`actions should create PLAY_FIRST_VIDEO when playing first video 1`] = `
Array [
  Object {
    "type": "PLAY_FIRST_VIDEO",
  },
]
`;

exports[`actions should create PLAY_NEXT_VIDEO when playing next video 1`] = `
Array [
  Object {
    "type": "PLAY_NEXT_VIDEO",
  },
]
`;

exports[`actions should create PLAY_PREVIOUS_VIDEO when playing previous video 1`] = `
Array [
  Object {
    "type": "PLAY_PREVIOUS_VIDEO",
  },
]
`;

exports[`actions should create PLAY_VIDEO when playing a video 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "PLAY_VIDEO",
  },
]
`;

exports[`actions should create PLAY_VIDEO when stopping a video 1`] = `
Array [
  Object {
    "payload": null,
    "type": "PLAY_VIDEO",
  },
]
`;

exports[`actions should create REMOVE_VIDEO when removing a video 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "REMOVE_VIDEO",
  },
]
`;

exports[`actions should create UPDATE_VIDEOS when updating videos 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "id": 1,
        "range": Array [
          0,
          242,
        ],
        "volume": 100,
        "youtubeId": "nfWlot6h_JM",
      },
      Object {
        "id": 2,
        "range": Array [
          0,
          256,
        ],
        "volume": 100,
        "youtubeId": "3tmd-ClpJxA",
      },
      Object {
        "id": 3,
        "range": Array [
          0,
          273,
        ],
        "volume": 100,
        "youtubeId": "e-ORhEE9VVg",
      },
      Object {
        "id": 4,
        "range": Array [
          0,
          245,
        ],
        "volume": 100,
        "youtubeId": "QcIy9NiNbmo",
      },
    ],
    "type": "UPDATE_VIDEOS",
  },
]
`;
